### PR TITLE
Save Client ID for registering GCM to that client

### PR DESCRIPTION
**Pull request by kunall17**
As in https://github.com/zulip/zulip-android/issues/112#issuecomment-236796320 This now fetches Google client ID and uses that one to register a GCM client ID

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@kunall17, it looks like you've already signed the Dropbox CLA. Thanks!

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 1]
```diff
@@ -237,8 +239,7 @@ protected void onCreate(Bundle savedInstanceState) {
             return;
         }
         this.logged_in = true;
-        notifications = new Notifications(this);
-        notifications.register();
+        registerGCM();
```
Comment (niftynei) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Needs to 'unregister' GCM after logout, including delete the saved google_client_id preference .

**Comment by niftynei** [comment]
looks pretty good, see comments in line.  In my test I didn't receive any notifications when I was pinged.

**Comment by niftynei** [comment]
Superceded by #123

**Pull request closed without merging**